### ISSUE TITLE
Add transactions from bank notifications

### ISSUE THREAD

**Issue by alex9099** [mtotschnig/MyExpenses#1497]
It would be nice if the app could listen to bank app notifications to automatically create transactions (based on the value and details that come in the notification).
Could also be nice to make it an option to just pre-fill data from the notification.

**Comment by mtotschnig** [comment]
I did not think this was possible, but after doing some research I learnt, that it is possible for the user to grant this permission.
Could you share a redacted example of such a notification, either here or to support@myexpenses.mobi?

**Comment by alex9099** [comment]
![ResizedImage_2024-06-02_22-19-12_1499](https://github.com/mtotschnig/MyExpenses/assets/7517045/631e22d8-7bf0-48c7-abcc-b839d71d1523)

Here's one example for payment, there are different notifications for transfers, payment (without card) and salary.
I'll update this comment (or make a new one) when I get more notifications as example

**Comment by Pietro395** [comment]
I think it's a great idea too, with the user's consent the app could go and save the transactions automatically and lìutente would just have to worry about categorizing them correctly.

Here are some examples of my notifications

Revolut:
![image](https://github.com/user-attachments/assets/462fda07-1a25-4d76-a867-52343f24d157)

Intesa San Paolo
![image](https://github.com/user-attachments/assets/a3e818c4-7d65-4d30-b438-420f425fe97b)



Perhaps one could use a regexp to locate the value and in case there are two values where one is the balance always take the lesser, as in the case of Revolut?

**Comment by Pietro395** [comment]
would it be possible to read only the notifications of the banks app?

```
Revolut - com.revolut.revolut

N26 - de.number26.android

Bunq - com.bunq.android

Monese - com.monese.monese.live

Monzo - co.uk.getmondo

Starling Bank - com.starlingbank.android

TransferWise (Wise) - com.transferwise.android

Banco Santander - com.santander.santanderUK

ING - com.ing.mobile

BBVA - com.bbva.bbvaapp

UniCredit - eu.unicredit.mobilebanking

Deutsche Bank - com.db.pbc.phototan

Crédit Agricole - com.creditagricole.androidapp

BNP Paribas - com.bnpp.easybank

Intesa Sanpaolo - com.intesasanpaolo.mobile

Raiffeisen Bank - com.raiffeisenbank.mobile

ABN AMRO - nl.abnamro.apps.pbankieren

HSBC - com.hsbc.mobilebanking

Barclays - com.barclays.android.barclaysmobilebanking

Lloyds Bank - com.grppl.android.shell.CMBlloydsTSB

NatWest - com.rbs.mobile.android.natwest

Royal Bank of Scotland (RBS) - com.rbs.mobile.android.rbs

CaixaBank - com.lacaixa.mobile.android.newwapicon

Société Générale - com.societegenerale.mobile.lappli

KBC Bank - com.kbc.mobile.android.phone.kbc

Erste Bank - at.spardat.netbanking

Commerzbank - de.commerzbanking.mobil

DZ Bank - de.dzbank.dzbankapp

DNB - no.dnb.vipps

Nordea - com.nordea.mobilebank

Swedbank - se.swedbank.mobil

SEB (Skandinaviska Enskilda Banken) - se.seb.mobile.android

Danske Bank - com.danskebank.mobilebank

Handelsbanken - com.handelsbanken.next

Alpha Bank - gr.alpha.mobilebanking

Eurobank Ergasias - gr.eurobank.mobile.app

Piraeus Bank - com.piraeusbank.winbank

Bank of Ireland - com.bankofireland.mobilebanking

Permanent TSB - com.permanenttsb.mobile

Banca Transilvania - ro.btrl.mbt

BRD - Groupe Société Générale - ro.brd

OTP Bank - hu.otp.mbanking

Sberbank Europe - com.sberbankmobile

Bank Millennium - pl.bankmillennium

PKO Bank Polski - pl.ipko

Alior Bank - pl.aliorbank.mbank

Banco BPM - it.bancopopolare

Banca MPS (Monte dei Paschi di Siena) - com.mps.mobilitas

FinecoBank - com.fineco

Hello bank! - com.bnpparibas.hellobank
```

**Comment by ItsIgnacioPortal** [comment]
> would it be possible to read only the notifications of the banks app?

No. Android permissions either allow reading all notifications, or none, through the permission `android.permission.BIND_NOTIFICATION_LISTENER_SERVICE`.

**Comment by ItsIgnacioPortal** [comment]
I've actually tackled this using [Tasker](https://tasker.joaoapps.com/). I've made an automation that detects the notifications sent by bank apps, and adds the corresponding transactions in MyExpenses. Additionally, for apps that don't issue notifications when a transaction occurs, I've made it so a floating bubble button shows up, and that button allows you to open MyExpenses in the "edit transaction" view, so it's a bit faster than just opening MyExpenses manually.

![](https://i.imgur.com/vNKJKMy.png)

The Tasker project is available at: https://taskernet.com/shares/?user=AS35m8mA86U3bLHr1n%2Flh2qwbZkFUhE5%2FlS7gUmcRAEWvhQ%2B55enof9KS2Rm0zEfRfcCnMpNPeFHtY8%3D&id=Project%3AMyExpenses+Automatic+Notification+Loader

Instructions for setting it up are included inside the project in the Task "SMP - INSTRUCTIONS", as well as in the taskernet page.
